website/docs/guides/version_4_upgrade.html.markdown

```diff
@@ -0,0 +1,295 @@
+  - That does not support versioning, see the local state instructions above
+  - That supports *versioning* you can revert the Terraform state file to a previous
+    version by hand. If you do so and Terraform created resources as part of a
+    `terraform apply`, you'll need to either `terraform import` them or delete
+    them by hand.
+
+## Upgrade Topics
+
+<!-- TOC depthFrom:2 depthTo:2 -->
+
+- [Provider Version Configuration](#provider-version-configuration)
+- [Provider Policy Changes](#provider-policy-changes)
+
+<!-- /TOC -->
+
+## Provider Version Configuration
+
+-> Before upgrading to version `4.0.0`, it is recommended to upgrade to the most
+recent version of the provider (`3.25.0`) and ensure that your environment
+successfully runs [`terraform plan`](https://www.terraform.io/docs/commands/plan.html)
+without unexpected changes or deprecation notices.
+
+It is recommended to use [version constraints](https://www.terraform.io/docs/configuration/providers.html#provider-versions)
+when configuring Terraform providers. If you are following that recommendation,
+update the version constraints in your Terraform configuration and run
+[`terraform init`](https://www.terraform.io/docs/commands/init.html) to download
+the new version.
+
+If you aren't using version constraints, you can use `terraform init -upgrade`
+in order to upgrade your provider to the latest released version.
+
+For example, given this previous configuration:
+
+```hcl
+provider "vault" {
+  # ... other configuration ...
+
+  version = "~> 3.25.0"
+}
+```
+
+An updated configuration:
+
+```hcl
+provider "vault" {
+  # ... other configuration ...
+
+  version = "~> 4.0.0"
+}
+```
+
+## Provider Policy Changes
+
+Version `4.0.0` of the Vault provider made changes to the underlying Vault API
+calls, which in turn may require policy adjustments in environments where
+permissions are least privilege.
+
+Please see the [Capabilities](https://developer.hashicorp.com/vault/docs/concepts/policies#capabilities)
+section of the Vault Policies documentation for more information on Vault
+policies.
+
+### Auth method resource changes
+
+The below table specifies what changed between version 3.X and 4.X for the
+following resources:
+
+#### Data sources
+- `vault_auth_backend`
+
+#### Resources
+- `vault_auth_backend`
+- `vault_gcp_auth_backend`
+- `vault_github_auth_backend`
+- `vault_jwt_auth_backend`
+- `vault_ldap_auth_backend`
+- `vault_okta_auth_backend`
+
+<table>
+<thead>
+  <tr>
+    <th colspan="2">3.X</th>
+    <th colspan="2">4.X</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>Method</td>
+    <td>Path</td>
+    <td>Method</td>
+    <td>Path</td>
+  </tr>
+  <tr>
+    <td>GET</td>
+    <td>sys/auth</td>
+    <td>GET</td>
+    <td>sys/auth/:path</td>
+  </tr>
+</tbody>
+</table>
+
+For example, in version 3.X the following policy is required for the
+`vault_gcp_auth_backend` resource to perform the GET `sys/auth` HTTP operation:
```

Comment:
We might want to say why these need to perform a GET on `sys/auth` (fetching mount metadata?) Just for illustration:

```suggestion
For example, in version 3.X the following policy is required for the
`vault_gcp_auth_backend` resource since its mount metadata is retrieved with the GET `sys/auth` HTTP operation:
```

Or since it's the same for all these resources we could state it in the intro for the section too.